Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanAlgorithm.cpp

```diff
@@ -174,73 +199,240 @@ void GalileanAlgorithm::InitializeSpectralCoefficients(const SpectralKSpace& spe
+                const Complex T2aux = amrex::exp(I * ckdt); // limit of T2 for nu=1
+
+                // See equation (12a)
+                C   (i,j,k) = std::cos(ckdt);
+                S_ck(i,j,k) = std::sin(ckdt) / ck;
+
+                // See equation (12b)
+                const Real    nu      = kv / ck;
+                const Complex th      = amrex::exp(  I * 0.5_rt * kv * dt);
+                const Complex th_star = amrex::exp(- I * 0.5_rt * kv * dt);
+
+                // This is exp(i*(k \dot v_gal)*dt)
+                T2(i,j,k) = th * th;
+
+                if ( (nu != 1.) && (nu != 0.) ) {
+
+                    // x1 is the coefficient chi_1 in equation (12c)
+                    Complex x1 = 1._rt / (1._rt - nu*nu)
+                        * (th_star - C(i,j,k) * th + I * kv * S_ck(i,j,k) * th);
+
+                    // X1 multiplies i*(k \times J) in the update equation for B
+                    X1(i,j,k) = th * x1 / (ep0 * c2 * k2);
+
+                    if (update_with_rho) {
+                        // X2 multiplies rho_new in the update equation for E
+                        // X3 multiplies rho_old in the update equation for E
+                        X2(i,j,k) = (x1 - th * (1._rt - C(i,j,k))) / (th_star - th) / (ep0 * k2);
+                        X3(i,j,k) = (x1 - th_star * (1._rt - C(i,j,k))) / (th_star - th) / (ep0 * k2);
+                    } else {
+                        // X2_old is the coefficient chi_2 in equation (12d)
+                        // X3_old is the coefficient chi_3 in equation (12d)
+                        // X2 multiplies (k \dot E) in the update equation for E
+                        // X3 multiplies (k \dot J) in the update equation for E
+                        X2_old = (x1 - th * (1._rt - C(i,j,k))) / (th_star - th);
+                        X3_old = (x1 - th_star * (1._rt - C(i,j,k))) / (th_star - th);
+                        X2(i,j,k) = T2(i,j,k) * (X2_old - X3_old) / k2;
+                        X3(i,j,k) = I * X2_old * (T2(i,j,k) - 1._rt) / (ep0 * k2 * kv);
+                    }
+                    // X4 multiplies J in the update equation for E
+                    X4(i,j,k) = I * kv * X1(i,j,k) - T2(i,j,k) * S_ck(i,j,k) / ep0;
                 }
-                if ( nu == 1.) {
-                    X1(i,j,k) = (1._rt - e_theta*e_theta + 2._rt*I*c*k_norm*dt) / (4._rt*c*c*ep0*k_norm*k_norm);
-                    X2(i,j,k) = (3._rt - 4._rt*e_theta + e_theta*e_theta + 2._rt*I*c*k_norm*dt) / (4._rt*ep0*k_norm*k_norm*(1._rt - e_theta));
-                    X3(i,j,k) = (3._rt - 2._rt/e_theta - 2._rt*e_theta + e_theta*e_theta - 2._rt*I*c*k_norm*dt) / (4._rt*ep0*(e_theta - 1._rt)*k_norm*k_norm);
-                    X4(i,j,k) = I*(-1._rt + e_theta*e_theta + 2._rt*I*c*k_norm*dt) / (4._rt*ep0*c*k_norm);
+                // Limits for nu=0
+                if (nu == 0.) {
+                    // X1 multiplies (k \times J) in the update equation for B
```

Comment:
In this and in one other comment, it seems “i*..." is missing?

Reply:
That's correct, thanks for catching it! Done in 33a7fb3081990fa600d2c63571cb279003eb58da.